deps/dicer/lib/Dicer.js

```diff
@@ -1,240 +1,206 @@
-var WritableStream = require('stream').Writable,
-    inherits = require('util').inherits;
-
-var StreamSearch = require('../../streamsearch/sbmh');
-
-var PartStream = require('./PartStream'),
-    HeaderParser = require('./HeaderParser');
-
-var DASH = 45,
-    B_ONEDASH = Buffer.from('-'),
-    B_CRLF = Buffer.from('\r\n'),
-    EMPTY_FN = function() {};
-
-function Dicer(cfg) {
-  if (!(this instanceof Dicer))
-    return new Dicer(cfg);
-  WritableStream.call(this, cfg);
-
-  if (!cfg || (!cfg.headerFirst && typeof cfg.boundary !== 'string'))
-    throw new TypeError('Boundary required');
-
-  if (typeof cfg.boundary === 'string')
-    this.setBoundary(cfg.boundary);
-  else
-    this._bparser = undefined;
-
-  this._headerFirst = cfg.headerFirst;
-
-  var self = this;
-
-  this._dashes = 0;
-  this._parts = 0;
-  this._finished = false;
-  this._realFinish = false;
-  this._isPreamble = true;
-  this._justMatched = false;
-  this._firstWrite = true;
-  this._inHeader = true;
-  this._part = undefined;
-  this._cb = undefined;
-  this._ignoreData = false;
-  this._partOpts = (typeof cfg.partHwm === 'number'
-                    ? { highWaterMark: cfg.partHwm }
-                    : {});
-  this._pause = false;
+const WritableStream = require('stream').Writable
+const inherits = require('util').inherits
+
+const StreamSearch = require('../../streamsearch/sbmh')
+
+const PartStream = require('./PartStream')
+const HeaderParser = require('./HeaderParser')
+
+const DASH = 45
+const B_ONEDASH = Buffer.from('-')
+const B_CRLF = Buffer.from('\r\n')
+const EMPTY_FN = function () {}
+
+function Dicer (cfg) {
+  if (!(this instanceof Dicer)) { return new Dicer(cfg) }
+  WritableStream.call(this, cfg)
+
+  if (!cfg || (!cfg.headerFirst && typeof cfg.boundary !== 'string')) { throw new TypeError('Boundary required') }
 
-  this._hparser = new HeaderParser(cfg);
-  this._hparser.on('header', function(header) {
-    self._inHeader = false;
-    self._part.emit('header', header);
-  });
+  if (typeof cfg.boundary === 'string') { this.setBoundary(cfg.boundary) } else { this._bparser = undefined }
 
+  this._headerFirst = cfg.headerFirst
+
+  const self = this
+
+  this._dashes = 0
+  this._parts = 0
+  this._finished = false
+  this._realFinish = false
+  this._isPreamble = true
+  this._justMatched = false
+  this._firstWrite = true
+  this._inHeader = true
+  this._part = undefined
+  this._cb = undefined
+  this._ignoreData = false
+  this._partOpts = (typeof cfg.partHwm === 'number'
+    ? { highWaterMark: cfg.partHwm }
+    : {})
+  this._pause = false
+
+  this._hparser = new HeaderParser(cfg)
+  this._hparser.on('header', function (header) {
+    self._inHeader = false
+    self._part.emit('header', header)
+  })
 }
-inherits(Dicer, WritableStream);
+inherits(Dicer, WritableStream)
```

Comment:
we should try rewriting to classes and measure performance

Reply:
yeah, maybe for the next release :)